translate.c

```diff
@@ -111,6 +128,23 @@ static int translate_readonly(const char *file)
 		}
 		firmware_value[n] = '\0';
 
+		/* first try path from sysfs */
+		if ((strlen(fw_sysfs_path) > 0) &&
+		    (strlen(fw_sysfs_path) + 1 +strlen(firmware_value) + 1 + strlen(file) + 1 < sizeof(path))) {
+			strcpy(path, fw_sysfs_path);
+			strcat(path, "/");
+			strcat(path, dirname(firmware_value));
+			strcat(path, "/");
+			strcat(path, file);
```

Comment:
This looks like a perfect usecase for `asprintf`

Reply:
asprintf/vanprintf is non-standard GNU/BSD extension and existing code around already uses strcpy/strcat everywhere. I can try though to use it here if you insist

Reply:
No, I don't.